flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/connector/sink/SinkFunctionProvider.java

```diff
@@ -20,19 +20,39 @@
 
 import org.apache.flink.annotation.PublicEvolving;
 import org.apache.flink.streaming.api.functions.sink.SinkFunction;
+import org.apache.flink.table.connector.ParallelismProvider;
 import org.apache.flink.table.data.RowData;
 
+import java.util.Optional;
+
 /**
  * Provider of a {@link SinkFunction} instance as a runtime implementation for {@link DynamicTableSink}.
  */
 @PublicEvolving
-public interface SinkFunctionProvider extends DynamicTableSink.SinkRuntimeProvider {
+public interface SinkFunctionProvider extends DynamicTableSink.SinkRuntimeProvider, ParallelismProvider {
 
 	/**
 	 * Helper method for creating a static provider.
 	 */
 	static SinkFunctionProvider of(SinkFunction<RowData> sinkFunction) {
-		return () -> sinkFunction;
+		return of(sinkFunction, Optional.empty());
+	}
+
+	/**
+	 * Helper method for creating a static provider, sink parallelism will be configured if non-empty parallelism is passed in.
+	 */
+	static SinkFunctionProvider of(SinkFunction<RowData> sinkFunction, Optional<Integer> parallelism) {
```

Comment:
I think we don't need provide method.
https://flink.apache.org/contributing/code-style-and-quality-java.html#java-optional
In Flink code style, it is recommended to use the Optional only in method return values.

Reply:
@JingsongLi 
> it is recommended to use the Optional only in method return values 
copy that.
> I think we don't need provide method.
Well, since `SinkFunctionProvider ` implements `ParallelismProvider` as default, as far as I’m concerned, there should be a method passing the parallelism in. Or Is there an better alternative? It‘s kind of u to tell me that~

Reply:
For the user which want to set parallelism, I think it is OK to let them create an implementation class.